dashboard/src/modules/components/OverviewComponent/SavedRunsComponent.jsx

```diff
@@ -0,0 +1,186 @@
+
+  /* Selecting */
+  const areAllRunsSelected =
+    savedRuns?.length > 0 && savedRuns?.length === selectedRuns?.length;
+  const selectAllRuns = (isSelecting) => {
+    dispatch(setSelectedRuns(isSelecting ? savedRuns.map((r) => r) : []));
+  };
+  const onSelectRuns = (run, __rowIndex, isSelecting) => {
+    const otherSelectedRuns = selectedRuns.filter(
+      (r) => r.resource_id !== run.resource_id
+    );
+    const c = isSelecting ? [...otherSelectedRuns, run] : otherSelectedRuns;
+    dispatch(setSelectedRuns(c));
+  };
+  const isRowSelected = (run) =>
+    selectedRuns.filter((item) => item.name === run.name).length > 0;
+  /* Selecting */
+
+  /* Actions Row */
+  const moreActionItems = (dataset) => [
+    {
+      title:
+        dataset.metadata["dataset.access"] === "public"
+          ? "Publish"
+          : "Unpublish",
+      onClick: () => {
+        const accessType =
+          dataset.metadata["dataset.access"] === "public"
+            ? "private"
+            : "public";
+        dispatch(publishDataset(dataset, accessType));
+      },
+    },
+    {
+      title: dataset.metadata["global.dashboard.seen"]
+        ? "Mark unread"
+        : "Mark read",
+      onClick: () =>
+        dispatch(
+          updateDataset(
+            dataset,
+            "read",
+            !dataset.metadata["global.dashboard.seen"]
+          )
+        ),
+    },
+    {
+      title: "Delete",
+      onClick: () => dispatch(deleteDataset(dataset)),
+    },
+  ];
+  /* Actions Row */
+  const makeFavorites = (dataset, isFavoriting = true) => {
+    dispatch(updateDataset(dataset, "favorite", isFavoriting));
+  };
+  const calculatePercent = (dateStr) => {
+    const daysLeft = findNoOfDays(dateStr);
+    return Math.abs(daysLeft / 1000) * 100;
+  };
+  const columnNames = {
+    result: "Result",
+    createdtime: "Created Time",
+    scheduled: "Scheduled for deletion on",
+    access: "Access",
+  };
+  return (
+    <div className="savedRuns-table-container">
+      <OuterScrollContainer>
+        <InnerScrollContainer>
+          <TableComposable isStickyHeader variant={"compact"}>
+            <Thead>
+              <Tr>
+                <Th
+                  select={{
+                    onSelect: (_event, isSelecting) =>
+                      selectAllRuns(isSelecting),
+                    isSelected: areAllRunsSelected,
+                  }}
+                ></Th>
+                <Th>{columnNames.result}</Th>
+                <Th>{columnNames.createdtime}</Th>
+                <Th>{columnNames.scheduled}</Th>
+                <Th>{columnNames.access}</Th>
+                <Th></Th>
+                <Th></Th>
+              </Tr>
+            </Thead>
+            {savedRuns.map((item, rowIndex) => {
+              const rowActions = moreActionItems(item);
+              const isItemFavorited = !!item?.metadata?.["user.favorite"];
+              const isItemSeen = !!item?.metadata?.["dashboard.seen"];
```

Comment:
s/"dashboard/"global.dashboard/ ?

Reply:
This amplifies a comment I made yesterday but forgot to commit: for #2951 I tried adding `const` symbols for the metadata key names used in the dashboard. If these were centralized, it'd avoid "typos like this" ( 😆 ), which might be hard to find during review, from getting into `main`. The `user.favorite` above, while it violates the new "recommended sub-namespace standard", would work, but this would give you a `BAD_REQUEST` error when it runs.

Reply:
So... _also_ s/"user/"user.dashboard/ ?

(\*sigh\*, I didn't think to look for that one....)

Reply:
Created constants